aiida/cmdline/commands/cmd_import.py

```diff
@@ -157,14 +215,64 @@ def cmd_import(archives, webpages, group, extras_mode_existing, extras_mode_new,
+                        "Note: A temporary local file will be generated.".format(archive),
+                        default=manual_migration,
+                        abort=True)
             except Exception:
                 echo.echo_error('an exception occurred while importing the archive {}'.format(archive))
                 echo.echo(traceback.format_exc())
-                click.confirm('do you want to continue?', abort=True)
+                if not non_interactive:
+                    click.confirm('do you want to continue?', abort=True)
             else:
                 echo.echo_success('imported archive {}'.format(archive))
+
+            # Migrate archive if needed and desired
+            if migrate_archive:
+                migrate_archive = False
+
+                temp_out_file = 'migrated_importfile.aiida'
+
+                echo.echo_info("migrating archive {}".format(archive))
+
+                try:
+                    ctx.invoke(
+                        migrate,
+                        input_file=temp_folder.get_abs_path(temp_file),
+                        output_file=temp_folder.get_abs_path(temp_out_file),
+                        silent=False)
+                except Exception:
+                    echo.echo_error(
+                        "an exception occurred while migrating the archive {}.\n"
+                        "Download the archive file and run 'verdi export migrate' to update it.".format(archive))
+                    echo.echo(traceback.format_exc())
+                    if not non_interactive:
+                        click.confirm('do you want to continue?', abort=True)
+                else:
+                    echo.echo_success("archive migrated, proceeding with import")
+
+                try:
+                    import_data(
+                        temp_folder.get_abs_path(temp_out_file),
+                        group,
+                        extras_mode_existing=ExtrasImportCode[extras_mode_existing].value,
+                        extras_mode_new=extras_mode_new,
+                        comment_mode=comment_mode)
+                except exceptions.IncompatibleArchiveVersionError as exception:
+                    crit_message = "{} has been migrated, but it still cannot be imported.\n" \
+                        'Download the archive file and run `verdi export migrate` to update it.\n{}'.format(
+                        archive, exception)
+                    echo.echo_critical(crit_message)
+                except Exception:
+                    echo.echo_error('an exception occurred while importing the archive {}'.format(archive))
+                    echo.echo(traceback.format_exc())
+                    if not non_interactive:
+                        click.confirm('do you want to continue?', abort=True)
+                else:
+                    echo.echo_success('imported archive {} after migration'.format(archive))
```

Comment:
will it also go here, if no migration was needed?

Reply:
It shouldn't, since all this is under the `if migrate_archive:`

Reply:
Ok, I see now that there is a lot of code duplication for url vs local archives, which was perhaps partially already there before but is now growing even more because of the migration.

Can we essentially get rid of that? There isn't really much difference in the two code paths, right?

Reply:
I agree with this sentiment. I can try. The main difference is when to use the SandboxFolder.
This will however have to wait until tomorrow at the earliest, probably not until beginning of next week.

Reply:
Done, but not yet completely tested.

Reply:
Done.